apis/datadoghq/v1alpha1/datadogautoscaling_types.go

```diff
@@ -0,0 +1,282 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package v1alpha1
+
+import (
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// spec:
+//   targetRef:
+//     apiVersion: apps/v1
+//     kind: Deployment
+//     name: test
+//   owner: local
+//   policy:
+//     update:
+//       mode: Auto|DryRun
+//   recommender:
+//     name: containerMetrics
+//     containerMetrics:
+//       cpuUtilizationTarget: 60
+//   constraints:
+//     minReplicas: 1
+//     maxReplicas: 10
+//     containers:
+//       - name: "*"
+//         enabled: true
```

Comment:
I assume the purpose of this field is to factor in/out certain containers from recommendations? Does it apply to both vertical + horizontal dimensions?

Reply:
Yep that's what got me confused because I was thinking of an "inverted" HPA `ContainerResource` metric type for a moment.

Reply:
It uses the include/exclude syntax from VPA/MPA. Should we consider that Vertical/Horizontal share the same settings about included/excluded containers? I don't think it's necessary the case, but I'm not sure where to put the equivalent selected containers for HPA, as if we think about the current `DatadogMetrics` with arbitrary query, it does not require any information.